lib/charms/observability_libs/v0/kubernetes_compute_resources_patch.py

```diff
@@ -422,6 +440,9 @@ class KubernetesComputeResourcesPatch(Object):
     """A utility for patching the Kubernetes compute resources set up by Juju."""
 
     on = K8sResourcePatchEvents()  # pyright: ignore
+    PATCH_RETRY_STOP = tenacity.stop_after_delay(60 * 3)
+    PATCH_RETRY_WAIT = tenacity.wait_fixed(30)
```

Comment:
If the charm sleeps (does not return) for 30sec, then juju will conclude the agent is down and will elect another leader. This is not a good thing. You can experiment with e.g. a 31 sec sleep to see the exact cascade.
In fact you could write a discourse post about your findings, I'm sure many will find it interesting.

Reply:
Will verify the above as a separate activity from this PR. for the purpose of this PR, set the retry value to `20` seconds to avoid any possible risks.